mesonbuild/compilers/c.py

```diff
@@ -0,0 +1,996 @@
+# Copyright 2012-2017 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess, os.path
+
+from .. import mlog
+from .. import coredata
+from ..mesonlib import EnvironmentException, version_compare, Popen_safe
+
+from .compilers import *
+from . import Compiler
+from . import GnuCompiler
+from . import ClangCompiler
+from . import IntelCompiler
+
+class CCompiler(Compiler):
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        # If a child ObjC or CPP class has already set it, don't set it ourselves
+        if not hasattr(self, 'language'):
+            self.language = 'c'
+        super().__init__(exelist, version)
+        self.id = 'unknown'
+        self.is_cross = is_cross
+        self.can_compile_suffixes.add('h')
+        if isinstance(exe_wrapper, str):
+            self.exe_wrapper = [exe_wrapper]
+        else:
+            self.exe_wrapper = exe_wrapper
+
+    def needs_static_linker(self):
+        return True # When compiling static libraries, so yes.
+
+    def get_always_args(self):
+        '''
+        Args that are always-on for all C compilers other than MSVC
+        '''
+        return ['-pipe'] + get_largefile_args(self)
```

Comment:
__[flake8]__
* [F405] 'get_largefile_args' may be undefined, or defined from star imports: .compilers

<!-- Comment by [SideCI](https://sideci.com) -->